Class reference is failed when include 'or' at a package name

If package name includes "or" (e.g. "jp.or.example"), the `Ognl#parseExpression(String)` is failed as follow:

* Code

```java
    public void test_class_reference()
        throws Throwable
    {
        OgnlContext context = (OgnlContext) Ognl.createDefaultContext(null, new DefaultMemberAccess(false));
        {
            Node expr = (Node) Ognl.parseExpression("@java.util.UUID@randomUUID()"); // It's OK
            Object v = expr.getValue(context, null);
            System.out.println(v);
        }
        {
            Node expr = (Node) Ognl.parseExpression("@jp.or.example.IdUtils@generateId()"); // It's failed on parsing phase
            Object v = expr.getValue(context, null);
            System.out.println(v);
        }
    }
```

* Stack Trace

```
ognl.ExpressionSyntaxException: Malformed OGNL expression: @jp.or.example.IdUtils@generateId()
 [ognl.ParseException: Encountered " "or" "or "" at line 1, column 5.
Was expecting:
    <IDENT> ...
    ]
	at ognl.Ognl.parseExpression(Ognl.java:179)
	at org.ognl.test.enhance.TestExpressionCompiler.test_class_reference(TestExpressionCompiler.java:55)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at junit.framework.TestCase.runTest(TestCase.java:176)
	at junit.framework.TestCase.runBare(TestCase.java:141)
	at junit.framework.TestResult$1.protect(TestResult.java:122)
	at junit.framework.TestResult.runProtected(TestResult.java:142)
	at junit.framework.TestResult.run(TestResult.java:125)
	at junit.framework.TestCase.run(TestCase.java:129)
	at junit.framework.TestSuite.runTest(TestSuite.java:252)
	at junit.framework.TestSuite.run(TestSuite.java:247)
	at org.junit.internal.runners.JUnit38ClassRunner.run(JUnit38ClassRunner.java:86)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:68)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
Caused by: ognl.ParseException: Encountered " "or" "or "" at line 1, column 5.
Was expecting:
    <IDENT> ...
    
	at ognl.OgnlParser.generateParseException(OgnlParser.java:3232)
	at ognl.OgnlParser.jj_consume_token(OgnlParser.java:3098)
	at ognl.OgnlParser.className(OgnlParser.java:1822)
	at ognl.OgnlParser.classReference(OgnlParser.java:1795)
	at ognl.OgnlParser.staticReference(OgnlParser.java:1762)
```

--- a/src/main/javacc/ognl.jj
+++ b/src/main/javacc/ognl.jj
@@ -946,7 +946,7 @@ void unaryExpression() : {
         navigationChain()
         [
             "instanceof"
-            t = <IDENT>/*@bgen(jjtree) #Instanceof( 1) */
+            t = classNamePart()/*@bgen(jjtree) #Instanceof( 1) */
                          {
                            ASTInstanceof jjtn004 = new ASTInstanceof(JJTINSTANCEOF);
                            boolean jjtc004 = true;
@@ -965,7 +965,7 @@ void unaryExpression() : {
                            }
                          }
 /*@egen*/
-            (   "." t = <IDENT>         { sb.append('.').append( t.image ); }
+            (   "." t = classNamePart()         { sb.append('.').append( t.image ); }
             )*                          { ionode.setTargetType( new String(sb) ); }
         ]
     )
@@ -1376,9 +1376,45 @@ String className(): {
     StringBuffer result;
 }
 {
-    t=<IDENT>               { result = new StringBuffer( t.image ); }
-    ( "." t=<IDENT>         { result.append('.').append( t.image ); }
+    t=classNamePart()               { result = new StringBuffer( t.image ); }
+    ( "." t=classNamePart()         { result.append('.').append( t.image ); }
     )*                      { return new String(result); }
+}
+
+/**
+ * Helper production to parse class/package name parts that may include reserved keywords.
+ * This allows package names like "jp.or.example" where "or" is a reserved OGNL keyword.
+ * Returns a Token representing either an identifier or a reserved keyword used as an identifier.
+ */
+Token classNamePart(): {
+    Token t;
+}
+{
+    (
+        t=<IDENT>
+        | "or" { t = token; }
+        | "and" { t = token; }
+        | "not" { t = token; }
+        | "in" { t = token; }
+        | "bor" { t = token; }
+        | "xor" { t = token; }
+        | "band" { t = token; }
+        | "eq" { t = token; }
+        | "neq" { t = token; }
+        | "lt" { t = token; }
+        | "lte" { t = token; }
+        | "gt" { t = token; }
+        | "gte" { t = token; }
+        | "shl" { t = token; }
+        | "shr" { t = token; }
+        | "ushr" { t = token; }
+        | "new" { t = token; }
+        | "true" { t = token; }
+        | "false" { t = token; }
+        | "null" { t = token; }
+        | "instanceof" { t = token; }
+    )
+    { return t; }
 }
 
 void constructorCall()       : {/*@bgen(jjtree) Ctor */

--- a/src/test/java/ognl/test/PackageKeywordTest.java
+++ b/src/test/java/ognl/test/PackageKeywordTest.java
@@ -1,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package ognl.test;
+
+import junit.framework.TestCase;
+import ognl.ExpressionSyntaxException;
+import ognl.Ognl;
+
+/**
+ * Tests for parsing class references containing reserved keywords in package names.
+ * This addresses Issue #103 where expressions like @jp.or.example.IdUtils@method()
+ * would fail to parse because "or" is a reserved keyword in OGNL.
+ */
+public class PackageKeywordTest extends TestCase {
+
+    /**
+     * Test that a package name containing "or" keyword can be parsed.
+     * This is a common pattern in Japanese domain names (.jp.or.).
+     */
+    public void testPackageNameWithOrKeyword() throws Exception {
+        try {
+            Object expression = Ognl.parseExpression("@jp.or.example.Utils@staticMethod()");
+            assertNotNull(expression);
+        } catch (ExpressionSyntaxException e) {
+            fail("Failed to parse expression with 'or' keyword in package name: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Test that a package name containing "and" keyword can be parsed.
+     */
+    public void testPackageNameWithAndKeyword() throws Exception {
+        try {
+            Object expression = Ognl.parseExpression("@com.and.example.Utils@staticMethod()");
+            assertNotNull(expression);
+        } catch (ExpressionSyntaxException e) {
+            fail("Failed to parse expression with 'and' keyword in package name: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Test that a package name containing "not" keyword can be parsed.
+     */
+    public void testPackageNameWithNotKeyword() throws Exception {
+        try {
+            Object expression = Ognl.parseExpression("@com.not.example.Utils@staticMethod()");
+            assertNotNull(expression);
+        } catch (ExpressionSyntaxException e) {
+            fail("Failed to parse expression with 'not' keyword in package name: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Test that a package name containing "in" keyword can be parsed.
+     */
+    public void testPackageNameWithInKeyword() throws Exception {
+        try {
+            Object expression = Ognl.parseExpression("@com.in.example.Utils@staticMethod()");
+            assertNotNull(expression);
+        } catch (ExpressionSyntaxException e) {
+            fail("Failed to parse expression with 'in' keyword in package name: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Test that a package name containing multiple reserved keywords can be parsed.
+     */
+    public void testPackageNameWithMultipleKeywords() throws Exception {
+        try {
+            Object expression = Ognl.parseExpression("@jp.or.in.and.Utils@staticMethod()");
+            assertNotNull(expression);
+        } catch (ExpressionSyntaxException e) {
+            fail("Failed to parse expression with multiple keywords in package name: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Test that a static field reference with keyword in package works.
+     */
+    public void testStaticFieldWithKeywordInPackage() throws Exception {
+        try {
+            Object expression = Ognl.parseExpression("@jp.or.example.Constants@FIELD_NAME");
+            assertNotNull(expression);
+        } catch (ExpressionSyntaxException e) {
+            fail("Failed to parse static field reference with 'or' keyword in package name: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Test that instanceof with keyword in class name works.
+     */
+    public void testInstanceofWithKeywordInClassName() throws Exception {
+        try {
+            Object expression = Ognl.parseExpression("value instanceof jp.or.example.MyClass");
+            assertNotNull(expression);
+        } catch (ExpressionSyntaxException e) {
+            fail("Failed to parse instanceof expression with 'or' keyword in class name: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Test that constructor call with keyword in package works.
+     */
+    public void testConstructorWithKeywordInPackage() throws Exception {
+        try {
+            Object expression = Ognl.parseExpression("new jp.or.example.MyClass()");
+            assertNotNull(expression);
+        } catch (ExpressionSyntaxException e) {
+            fail("Failed to parse constructor expression with 'or' keyword in package name: " + e.getMessage());
+        }
+    }
+
+    /**
+     * Test various other reserved keywords that might appear in package names.
+     */
+    public void testOtherReservedKeywordsInPackages() throws Exception {
+        String[] keywords = {"bor", "xor", "band", "eq", "neq", "lt", "lte", "gt", "gte", "shl", "shr", "ushr"};
+
+        for (String keyword : keywords) {
+            try {
+                String expressionStr = "@com." + keyword + ".example.Utils@method()";
+                Object expression = Ognl.parseExpression(expressionStr);
+                assertNotNull(expression);
+            } catch (ExpressionSyntaxException e) {
+                fail("Failed to parse expression with '" + keyword + "' keyword in package name: " + e.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Test that the standard java.util.UUID example still works.
+     */
+    public void testStandardJavaPackageStillWorks() throws Exception {
+        try {
+            Object expression = Ognl.parseExpression("@java.util.UUID@randomUUID()");
+            assertNotNull(expression);
+        } catch (ExpressionSyntaxException e) {
+            fail("Failed to parse standard Java package reference: " + e.getMessage());
+        }
+    }
+}

